### PR TITLE
St/merging webapp

### PR DESCRIPTION
**Pull request by showduhtung**

### CODE REVIEW

File: server/db/config.ts [hunk 1]
```diff
@@ -2,12 +2,24 @@ import dotenv from 'dotenv';
...
+const config: ConfigInterface = {
   development: {
-    database: process.env.DB_NAME,
-    username: process.env.DB_USER,
-    password: process.env.DB_PASS,
-    host: process.env.DB_HOST,
+    database: process.env.DB_NAME ?? '',
+    username: process.env.DB_USER ?? '',
+    password: process.env.DB_PASS ?? '',
+    host: process.env.DB_HOST ?? '',
```
Comment (hchiam) [inline server/db/config.ts:22]: cool! nullish coalescing operator

File: src/assets/tab_data.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+  "tabs": [
+    {
+      "name": "Main"
+    },
+    {
+      "name": "RE"
+    },
+    {
+      "name": "RE"
```
Comment (hchiam) [inline src/assets/tab_data.json]: RE 2?

File: server/routes/users.ts [hunk 3]
```diff
@@ -83,10 +83,11 @@ router.post('/users', certify, async (req: Request, res: Response, next) => {
   try {
     const { email, firstName, lastName, password, churchId } = req.body;
     const token = crypto.randomBytes(16).toString('hex');
-    const userExists: UserInstance = await db.User.findOne({
+    const userExists: UserInstance | null = await db.User.findOne({
       where: { email },
       attributes: ['id', 'email'],
     });
+    // const userExists = !!user;
```
Comment (hchiam) [inline server/routes/users.ts]: i think this line can be deleted. doesn't look like it's being saved for later

**Review by hchiam — CHANGES_REQUESTED** [review]

File: package.json [hunk 4]
```diff
@@ -65,39 +71,40 @@
...
+    "react-beautiful-dnd": "^13.0.0",
+    "react-contextmenu": "^2.14.0",
+    "react-dom": "^16.9.0",
     "react-navigation": "^4.3.9",
-    "react-redux": "^7.2.0",
+    "react-query": "^2.26.2",
+    "react-query-devtools": "^2.6.3",
+    "react-redux": "^7.2.2",
+    "react-router-dom": "^5.2.0",
+    "react-table": "^7.6.2",
```
Comment (CedThen) [inline package.json]: not using this anymore, along with "@types/react-table": "^7.0.25", line 60

File: server/db/models/Role.ts [hunk 5]
```diff
@@ -7,6 +7,7 @@ const RoleFactory = (
   DataTypes: Sequelize.DataTypes,
 ): Sequelize.Model<RoleInstance, RoleAttributes> => {
   const attributes: SequelizeAttributes<RoleAttributes> = {
+    id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
```
Comment (CedThen) [inline server/db/models/Role.ts:10]: how come we didn't include the ID's before? db was still populating them with primary key id's tho

File: server/routes/users.ts [hunk 6]
```diff
@@ -118,11 +119,11 @@ router.post('/users', certify, async (req: Request, res: Response, next) => {
       switch (true) {
```
Comment (CedThen) [inline server/routes/users.ts:118]: is this like a reverse switch statement? very interesting

File: src/components/Auth/Login.tsx [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+import CssBaseline from '@material-ui/core/CssBaseline';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Checkbox from '@material-ui/core/Checkbox';
+import Grid from '@material-ui/core/Grid';
+import Box from '@material-ui/core/Box';
+import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
+import Typography from '@material-ui/core/Typography';
+import Container from '@material-ui/core/Container';
+import CircularProgress from '@material-ui/core/CircularProgress';
+
```
Comment (CedThen) [inline src/components/Auth/Login.tsx:19]: this doesnt appear to be the most up-to-date version, i implemented ValidatedTextField for some of the inputs here

File: src/components/Main/Schedule/ScheduleContainer.tsx [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+      <div className={classes.bottomButtonContainer}>
+        <Button onClick={onAddServiceClick} className={classes.addNewServiceButton}>
+          <AddIcon height={50} width={50} />
+          <span>Add New Service</span>
+        </Button>
+      </div>
+    </div>
+  );
+
+  function closeDialogHandler(response: any) {
```
Comment (CedThen) [inline src/components/Main/Schedule/ScheduleContainer.tsx]: were u going to move these above the jsx return?

File: src/components/Main/Schedule/Table.tsx [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+  onTaskModified: any; // any
+}
+
+export const Table = ({ data, access, onTaskModified }: TableProps) => {
+  const outerRef = useRef(null);
+  const classes = useStyles();
+  // const [dataRows, setDataRows] = useState([...data]);
+
+  const { columns, services } = data;
+  console.log('data', data);
```
Comment (CedThen) [inline src/components/Main/Schedule/Table.tsx:33]: console.log here

File: src/components/Main/Schedule/Table.tsx [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+      <ContextMenu
+        outerRef={outerRef}
+        addRowHandler={insertRow}
+        deleteRowHandler={deleteRow}
+      />
+      <MaUTable className={classes.table} ref={outerRef}>
+        <TableHead>
+          <TableRow key="Column header">
+            {columns.map((column: any, index: number) => (
+              // eslint-disable-next-line react/no-array-index-key
```
Comment (CedThen) [inline src/components/Main/Schedule/Table.tsx:46]: what are these eslint comments for?

File: src/components/Main/Schedule/services.ts [hunk 11]
```diff
@@ -0,0 +1,112 @@
+import { DayIndexOptions } from '../../../shared/types';
```
Comment (CedThen) [inline src/components/Main/Schedule/services.ts:1]: the name made me think this file had to do with services, like from the database

File: src/assets/tab_data.json (hunk 2 above)
Comment (showduhtung) [reply src/assets/tab_data.json]: ha not a big deal. it's not even being used

File: server/db/models/Role.ts (hunk 5 above)
Comment (showduhtung) [reply server/db/models/Role.ts:10]: I honestly forget the exact reason why. This was done around the same time I was doing my scala project where I discovered this issue and read up on it. Long story short, better to implement it. It helps when referencing it.

File: server/routes/users.ts (hunk 6 above)
Comment (showduhtung) [reply server/routes/users.ts:118]: yup, I also used this in authentication. It's kinda a hack but not really as switch(true) was never technically meant to be used that way, but to me it's a helluva lot cleaner than a buttload of "if else" statements.

File: src/components/Main/Schedule/ScheduleContainer.tsx (hunk 8 above)
Comment (showduhtung) [reply src/components/Main/Schedule/ScheduleContainer.tsx]: oh yeah ty

File: src/components/Main/Schedule/Table.tsx (hunk 9 above)
Comment (hchiam) [reply src/components/Main/Schedule/Table.tsx:33]: Remember to remove the [console log](https://memegenerator.net/img/instances/68620658/a-wild-consolelog-appeared.jpg) if it's not needed.

File: src/components/Main/Schedule/Table.tsx (hunk 10 above)
Comment (showduhtung) [reply src/components/Main/Schedule/Table.tsx:46]: it says we shouldnt use the array index in key.

**Pull request merged**